sdk/planetarycomputer/azure-planetarycomputer/samples/planetary_computer_01_ingestion_management.py

```diff
@@ -0,0 +1,427 @@
+    # Generate a valid SAS token format with required fields: permission, start, and expiration
+    from datetime import datetime, timedelta
+
+    start_time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+    expiry_time = (datetime.utcnow() + timedelta(days=7)).strftime("%Y-%m-%dT%H:%M:%SZ")
+    updated_token = f"sp=rl&st={start_time}&se={expiry_time}&sv=2023-01-03&sr=c&sig=UpdatedRandomSignature123456"
+
+    updated_connection_info = SharedAccessSignatureTokenConnection(
+        container_uri=sas_container_uri, shared_access_signature_token=updated_token
+    )
+    updated_ingestion_source = SharedAccessSignatureTokenIngestionSource(
+        id=source_id, connection_info=updated_connection_info
+    )
+
+    logging.info("Second call to create_or_replace_source with updated SAS token")
+    second_result = client.ingestion.replace_source(id=source_id, body=updated_ingestion_source)
+    logging.info(f"Second call result: {second_result.id}")
+
+    return second_result.id
+
+
+def get_source_by_id(client: PlanetaryComputerProClient, source_id: str):
+    """Retrieve a specific ingestion source by ID.
+
+    This demonstrates using get_source to fetch a specific source directly
+    instead of listing all sources.
+    """
+    logging.info(f"Retrieving ingestion source: {source_id}")
+
+    try:
+        source = client.ingestion.get_source(id=source_id)
+        logging.info(f"Successfully retrieved source: {source.id}")
+        return source
+    except Exception as e:
+        logging.error(f"Failed to retrieve source {source_id}: {str(e)}")
+        return None
+
+
+def create_github_public_ingestion(client: PlanetaryComputerProClient, collection_id: str, source_catalog_url: str):
+    """Create, update, and run ingestion from sample public catalog on GitHub."""
+
+    # Delete all existing ingestions
+    logging.info("Deleting all existing ingestions...")
+    existing_ingestions = list(client.ingestion.list(collection_id=collection_id))
+    for ingestion in existing_ingestions:
+        client.ingestion.begin_delete(collection_id=collection_id, ingestion_id=ingestion.id, polling=True)
+        logging.info(f"Deleted existing ingestion: {ingestion.id}")
+
+    # Create ingestion definition
+    ingestion_definition = IngestionDefinition(
+        import_type=IngestionType.STATIC_CATALOG,
+        display_name="Sample Ingestion",
+        source_catalog_url=source_catalog_url,
+        keep_original_assets=True,
+        skip_existing_items=True,  # Skip items that already exist
+    )
+
+    # Create the ingestion
+    # TODO: The SDK's ingestion.create() method behavior should be reviewed.
+    # Currently, calling create() on an existing ingestion only performs updates rather than
+    # failing or replacing the ingestion. This should be clarified or a separate create_or_update
+    # method should be provided for idempotent operations.
```

Comment:
We should move this into an open Github issue to track it rather than a TODO in the public sample.